pkg/client/unversioned/helper.go

```diff
@@ -258,16 +258,35 @@ func SetKubernetesDefaults(config *restclient.Config) error {
 	if config.APIPath == "" {
 		config.APIPath = legacyAPIPath
 	}
-	g, err := registered.Group(api.GroupName)
-	if err != nil {
-		return err
+	if config.GroupVersion == nil || config.GroupVersion.Group != api.GroupName {
```

Comment:
Why are you doing this check?  Why is the core group different from other groups?